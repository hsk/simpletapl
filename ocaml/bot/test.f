/* Examples for testing */

lambda x:Top. x;
(lambda x:Top. x) (lambda x:Top. x);
(lambda x:Top->Top. x) (lambda x:Top. x);


lambda x:Bot. x;
lambda x:Bot. x x;

y:Bot->Bot;
x:Bot;
y x;
/* z; error */
