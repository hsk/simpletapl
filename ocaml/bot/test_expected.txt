lambda x:Top.x : Top -> Top
lambda x:Top.x : Top
lambda x:Top.x : Top -> Top
lambda x:Bot.x : Bot -> Bot
lambda x:Bot.x x : Bot -> Bot
y : Bot -> Bot
x : Bot
y x : Bot
